lib/src/lifecycle_module.dart

```diff
@@ -67,9 +67,9 @@ abstract class LifecycleModule {
   /// Calls the onLoad() method, which can be implemented on a Module.
   /// Executes the willLoad and didLoad event streams.
   Future load() async {
-    _willLoadController.add(this);
+    await _willLoadController.add(this);
     await onLoad();
-    _didLoadController.add(this);
+    await _didLoadController.add(this);
```

Comment:
My guess is that it works because the `await` syntax causes that line to be wrapped in another async block when compiled, which effectively waits an extra tick. But `StreamSink.add()` definitely returns void (https://api.dartlang.org/1.13.0/dart-async/EventSink/add.html) so adding an `await` isn't correct, it just produces a side effect that happens to work. For that reason, I'd vote for #41.